src/context/ApplicationContext.jsx

```diff
@@ -2,30 +2,41 @@ import { useState, useContext, createContext, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { UrlSearchParamBuilder } from '../utils/api-utils/UrlSearchParamBuilder';
 import { getTimePeriodTypes } from '../api/services/timecardService';
+import { validateServiceErrors } from '../utils/api-utils/ApiUtils';
 
 export const ApplicationContext = createContext();
 
-const fetchTimePeriodTypesData = async () => {
+const setTimePeriodTypesData = async (setTimePeriodTypes, setServiceError) => {
   const params = new UrlSearchParamBuilder()
     .setTenantId('00000000-0000-0000-0000-000000000000')
     .getUrlSearchParams();
-  const response = await getTimePeriodTypes(params);
-  return response.data?.items;
-  /* TODO: Error handling when server raises error, the above needs to be enclosed in a try catch block
-    and use error handling similar to similar to:
-    setSummaryErrors(error); */
+
+  validateServiceErrors(
+    setServiceError,
+    async () => {
+      const response = await getTimePeriodTypes(params);
+      setTimePeriodTypes(response.data?.items);
+    },
+    false
+  );
 };
 
 export const ApplicationProvider = ({ children }) => {
   const [timePeriodTypes, setTimePeriodTypes] = useState([]);
+  const [serviceError, setServiceError] = useState({
+    hasError: false,
+    recoverable: true,
+  });
 
   const value = {
     timePeriodTypes,
     setTimePeriodTypes,
+    serviceError,
+    setServiceError,
   };
 
   useEffect(async () => {
-    setTimePeriodTypes(await fetchTimePeriodTypesData());
+    setTimePeriodTypesData(setTimePeriodTypes, setServiceError);
```

Comment:
The `useEffect` might not need to be async anymore if we aren't awaiting this function

Reply:
good spot. removed